lib/Cro/HTTP/Client.pm6

```diff
@@ -539,6 +540,7 @@ class Cro::HTTP::Client {
         $request.append-header('Host', $url.host ~
             ($port && $port != 80 | 443 ?? ":$port" !! ""));
         if self {
+            $request.append-header('User-agent', $.user-agent) if $.user-agent && (%options<user-agent>:!exists);
```

Comment:
This should only happen if nothing in `!set-headers` results in a `User-agent` header.

Reply:
How does it change the logic?
If we work with an instance where user-agent was passed on instantiation and %options arg does not override it, we set the header, no?